nncf/experimental/post_training/algorithms/fast_bias_correction/algorithm.py

```diff
@@ -0,0 +1,223 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from audioop import bias
+from typing import Dict
+from typing import List
+from typing import TypeVar
+from typing import Union
+
+import numpy as np
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.common.utils.backend import BackendType, get_backend
+from nncf.common.graph.transformations.commands import TargetPoint
+from nncf.experimental.post_training.algorithms import AlgorithmParameters
+from nncf.experimental.post_training.algorithms.algorithm import Algorithm
+from nncf.experimental.post_training.algorithms.algorithm import PostTrainingAlgorithms
+from nncf.experimental.post_training.algorithms.fast_bias_correction.onnx_algo_backend import ONNXAlgoBackend
+from nncf.experimental.post_training.api.engine import Engine
+from nncf.experimental.post_training.factories import NNCFGraphFactory
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPoint
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+# pylint: disable=protected-access
+
+
+class FastBiasCorrectionParameters(AlgorithmParameters):
+    """
+    Base class of FastBiasCorrection parameters.
+    """
+
+    def __init__(self, number_samples: int = 100, threshold: float = 2.0):
+        self.number_samples = number_samples
+        self.threshold = threshold
+
+    def to_json(self) -> Dict[str, Union[str, float, int]]:
+        """
+        Serialize all FastBiasCorrection parameters to JSON.
+        """
+
+
+# pylint: disable = too-many-function-args
+class FastBiasCorrection(Algorithm):
+
+    def __init__(self, parameters: FastBiasCorrectionParameters):
+        super().__init__()
+        self.number_samples = parameters.number_samples
+        self.threshold = parameters.threshold
+        self.nncf_graph = None
+        self._target_points_to_correct = []  # type: List[TargetPoint]
+        self._channel_axis_by_types = {
+            'Conv': 1,
+            'Gemm': -1,
+        }
+        self._backend_entity = None
+    
+    def _set_backend_entity(self, model):
+        model_backend = get_backend(model)
+        if model_backend == BackendType.ONNX:
```

Comment:
I think we should create an interface, like `AlgoBackend`, which will contain all the backend-specific features that should be implemented to support the method in the backend.
We need also to have a registry with the available backends and provide an API inside every algo that will return the supported backends, like `algo.available_backends() -> List[Backend]`

Reply:
In general, should we have a common `AlgoBackend` or specific interface for each algorithm and per-backend implementations nearby?

Reply:
It should be per-algorithm.